kfs-core/src/main/java/edu/arizona/kfs/fp/document/DisbursementVoucherDocument.java

```diff
@@ -382,4 +391,116 @@ public String getRouteStatus() {
+        boolean success = processOffsetGeneralLedgerPendingEntry(sequenceHelper, glpeSourceDetail, explicitEntry, offsetEntry);
+
+        processTaxWithholdingGeneralLedgerPendingEntries(sequenceHelper, glpeSourceDetail, explicitEntry, offsetEntry);
+
+        LOG.debug("processGenerateGeneralLedgerPendingEntries(GeneralLedgerPendingEntrySourceDetail, GeneralLedgerPendingEntrySequenceHelper) - end");
+        return success;
+    }
+
+    @Override
+    public boolean customizeOffsetGeneralLedgerPendingEntry(GeneralLedgerPendingEntrySourceDetail accountingLine, GeneralLedgerPendingEntry explicitEntry, GeneralLedgerPendingEntry offsetEntry) {
+        boolean value = super.customizeOffsetGeneralLedgerPendingEntry(accountingLine, explicitEntry, offsetEntry);
+        String taxAccount = getTaxAccount();
+        if (offsetEntry.getAccountNumber().equals(taxAccount)) {
+            // Get the object code for the tax offsets from the parameter
+            String glpeOffsetObjectCode = getGlpeOffsetObjectCode();
+            SystemOptions options = getOptionsService().getOptions(explicitEntry.getUniversityFiscalYear());
+            offsetEntry.setFinancialObjectCode(glpeOffsetObjectCode);
+            offsetEntry.refreshReferenceObject(KFSPropertyConstants.FINANCIAL_OBJECT);
+            offsetEntry.setFinancialObjectTypeCode(options.getFinancialObjectTypeAssetsCd());
+            offsetEntry.refreshReferenceObject(KFSPropertyConstants.OBJECT_TYPE);
+        }
+        return value;
+    }
+
+    private void processTaxWithholdingGeneralLedgerPendingEntries(GeneralLedgerPendingEntrySequenceHelper sequenceHelper, GeneralLedgerPendingEntrySourceDetail glpeSourceDetail, GeneralLedgerPendingEntry explicitEntry, GeneralLedgerPendingEntry offsetEntry) {
+        KualiDecimal dvnraTaxAmount = SpringContext.getBean(DisbursementVoucherTaxService.class).getNonResidentAlienTaxAmount(this);
+        if (!KualiDecimal.ZERO.equals(dvnraTaxAmount)) {
+            // Don't process actual withholding entry
+            String taxAccount = getTaxAccount();
+            if (!offsetEntry.getAccountNumber().equals(taxAccount)) {
+                DisbursementVoucherNonResidentAlienTax dvnrat = ((DisbursementVoucherDocument) this).getDvNonResidentAlienTax();
+                BigDecimal amount = offsetEntry.getTransactionLedgerEntryAmount().bigDecimalValue();
+                BigDecimal taxPercentWhole = dvnrat.getFederalIncomeTaxPercent().add(dvnrat.getStateIncomeTaxPercent()).bigDecimalValue();
+                BigDecimal taxPercent = taxPercentWhole.divide(new BigDecimal(100), 5, BigDecimal.ROUND_HALF_UP);
+                KualiDecimal withholdingAmount = new KualiDecimal(amount.multiply(taxPercent).setScale(KualiDecimal.SCALE, KualiDecimal.ROUND_BEHAVIOR));
+                KualiDecimal remitAmount = new KualiDecimal(amount).subtract(withholdingAmount);
+
+                if ((withholdingAmount != null) && KualiDecimal.ZERO.compareTo(withholdingAmount) != 0) {

```

Comment:
`(withholdingAmount != null)` is always true -- it's worth pointing out, since a null check can imply this could be null. As written, this could set the expectation for the next dev that this could be something it can't be. (I.e., if line 443 bombs, we exit this method, thus will never be null if we get here.)